mitx_residential_etl.py

```diff
@@ -0,0 +1,179 @@
+        os.makedirs(settings['Paths']['csv_folder'] + date_suffix + '/')
+        logger.info("csv folder(s) created")
+
+
+def get_course_ids():
+    """
+    Get a list of course ids that is necessary for the rest of the
+    functions to work.
+    """
+    global course_ids
+    dump_course_ids = subprocess.Popen(['/edx/bin/python.edxapp',
+                                        '/edx/app/edxapp/edx-platform/manage.py',
+                                        'lms', '--settings', 'aws',
+                                        'dump_course_ids'], stdout=subprocess.PIPE)
+    course_ids = dump_course_ids.communicate()[0].split()
+    return course_ids
+
+
+def export_course(course_ids):
+    for course_id in course_ids:
+        with subprocess.Popen(['/edx/bin/python.edxapp',
+                               '/edx/app/edxapp/edx-platform/manage.py',
+                               'lms', '--settings', 'aws',
+                               'export_course', course_id, course_id + b'.tar.gz'], stdout=subprocess.PIPE) as proc:
+            logger.info(proc.stdout.read())
+
+
+def get_list_of_staff():
+    pass
+
+
+def mysql_query(course_ids):
+    engine = create_engine('mysql+mysqlconnector://{}:{}@{}/{}'
+                           .format(mysql_creds_user, mysql_creds_pass, mysql_host, mysql_db))
+    connection = engine.connect()
+    for course_id in course_ids:
+        for key, value in query_dict.items():
+            query_text = text(value)
+            query = connection.execute(query_text, course_id=course_id.decode('utf8'))
+            write_csv(query, key)
+
+
+def write_csv(query, key):
+    with open(daily_folder + str(key) + '.csv', 'a+') as f:
+        writer = csv.writer(f)
+        for row in query:
+            writer.writerow(row)
+    f.close()
+
+
+def sync_to_s3(daily_folder, s3_bucket_name):
+    """
+    Sync local files to specified S3 bucket
+
+    Args:
+      daily_folder (str): folder containing msql query results.
+      s3_bucket_name (str): s3 bucket name
+    """
+    try:
+        cmd_output = subprocess.run(["aws", "s3", "sync", daily_folder,
+                                    "s3://" + s3_bucket_name + '/' + date_suffix],
+                                    check=True, stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+    except SyntaxError as err:
+        logger.exception("Failed to run sync command. Check if awscli is installed")
+        notify_slack_channel(f"Failed to run sync command: `{err}`")
+        sys.exit("[-] Failed to run sync command. Check if awscli is installed")
+    except subprocess.SubprocessError as err:
+        logger.exception("Failed to sync local files to s3 bucket")
+        notify_slack_channel(f"Sync failed: `{err}`")
+        sys.exit("[-] Failed to sync daily_folder to s3 bucket")
+    logger.info("S3 sync successfully ran: {}", cmd_output)
+    notify_slack_channel(f"Sync succeeded: `str({cmd_output}`)")
```

Comment:
One thing to note with f strings is that Ubuntu installs 3.5 from the system repositories which doesn't have that feature. We can use it, but it will require installing Python 3.6 on the target host.

Reply:
Yeah I saw that it needed 3.6 but think it might be easier to use 3.5 as after testing it took some finagling to get 3.6 with pip3 to work and use that instead of 3.5.x. Will change in new commit.